boot/freeldr/freeldr/arch/i386/hwdisk.c

```diff
@@ -139,10 +139,24 @@ DiskRead(ULONG FileId, VOID* Buffer, ULONG N, ULONG* Count)
     BOOLEAN ret;
     ULONGLONG SectorOffset;
 
+    if (DiskReadBufferSize == 0)
```

Comment:
Can this actually happen? If it can, is that specific to Xbox?
See (default) initialization at
https://git.reactos.org/?p=reactos.git;a=blob;f=boot/freeldr/freeldr/arch/i386/pcmem.c;hb=34bf24756a6ebad1bac8954670a30570cd2f7479#l611
Would an `ASSERT()` be more relevant?

Reply:
Looks like yes, it's specific to `xboxmem.c`. Agree about `ASSERT()`.